### ISSUE TITLE
Assign data in onPreResponse lifecycle extension

### ISSUE THREAD

**Issue by suhanw** [hapijs/yar#144]
#### Support plan

* *which support plan is this issue covered by?* (e.g. Community, Core, Plus, or Enterprise): **Community**
* *is this issue currently blocking your project?* (yes/no): **no**
* *is this issue affecting a production system?* (yes/no): **yes**

#### Context

* *node version*: **12.14.0**
* *module version*: **yar@9.1.0, hapi@18.1.0**
* *environment* (e.g. node, browser, native): **node**
* *used with* (e.g. hapi application, another framework, standalone, ...): **hapi application**
* *any other relevant information*: **n/a**

#### How can we help?

Hi! I'm fairly new to `yar` and `hapi`, but let me do my best to explain what I'm trying to accomplish: 
- My Hapi server is making calls to another API service, and I use `keepAlive` agent on these calls, but the challenge we are facing now is every once in a while the service closes the connection and responds with `ECONNRESET` error. 
- So our 'band-aid' solution for now is to detect these `ECONNRESET` errors, redirect to homepage, and show a simple alert. 
- What I was hoping to do is to (a) detect the `ECONNRESET` response on the `onPreResponse` lifecycle, (b) set an alert message in session, and (c) change the response to a redirect to homepage. Here's the snippet:
```js
server.ext('onPreResponse', function (request, h){
    let response = request.response;

    if (response.message && typeof response.message.includes === 'function' && response.message.includes('ECONNRESET')) {
        request.yar.set('alerts', `Uh-oh. Something went wrong on our end. Please try again.`);
        return h.response().takeover().redirect('/');
    }

    // other code
}
```

- Then, I was going to retrieve the alert message from session in one of my homepage route pre-handlers, and save the alert into an app state object that gets sent to the front-end to display the alert. 
```js
server.route({
    method: 'GET',
    path: '/',
    config: {
        pre: [
            // other pre-handlers
            { method: getBase, assign: 'basePayload' },
            // other pre-handlers
        ],
        handler: Methods.render
    }
});


const getBase = async (request, h) => {
    let alerts = request.yar.get('alerts');

    // other code
}
```

But for some reason, the `alerts` in session did not persist from `onPreResponse` to the homepage pre-handler, i.e., `request.yar.get('alerts') === null` in the pre-handler. Is there something about `onPreResponse` lifecycle that doesn't allow me to save data into session? Any help is greatly appreciated. 

Below are my `yar` options. I tried to distill the code snippets to just the relevant parts, but let me know if I can provide any further info that might help. 
```js
{
    plugin: require('yar'),
    options: {
        name: nconf.get('redis:cookieName'),
        maxCookieSize: 0,
        cache: {
            expiresIn: nconf.get('redis:ttl')
        },
        cookieOptions: {
            ttl: nconf.get('redis:ttl'),
            password: nconf.get('redis:secret'),
            isSecure:
                process.env.NODE_ENV === 'production' ||
                process.env.NODE_ENV === 'staging' ||
                process.env.NODE_ENV === 'ephemeral',
            isHttpOnly:
                process.env.NODE_ENV === 'production' ||
                process.env.NODE_ENV === 'staging' ||
                process.env.NODE_ENV === 'ephemeral'
        },
        storeBlank: false
    }
},
```
Many thanks!

**Comment by hueniverse** [comment]
What is happening first, yar registration or your custom onPreResponse ext setup?

**Comment by suhanw** [comment]
They are both plugins included in the same array, but `yar` appears first in the array: 
```js
const plugins = [
    // other plugins
    {
        plugin: require('yar'),
        options: {
            name: nconf.get('redis:cookieName'),
            maxCookieSize: 0,
            cache: {
                expiresIn: nconf.get('redis:ttl')
            },
            cookieOptions: {
                ttl: nconf.get('redis:ttl'),
                password: nconf.get('redis:secret'),
                isSecure:
                    process.env.NODE_ENV === 'production' ||
                    process.env.NODE_ENV === 'staging' ||
                    process.env.NODE_ENV === 'ephemeral',
                isHttpOnly:
                    process.env.NODE_ENV === 'production' ||
                    process.env.NODE_ENV === 'staging' ||
                    process.env.NODE_ENV === 'ephemeral'
            },
            storeBlank: false
        }
    },
    // other plugins
    { plugin: require('./routes/main') },
    // other plugins
];

await server.register(plugins)
```
`require('./routes/main')` is the plugin where the `onPreResponse` ext setup lives.

**Comment by hueniverse** [comment]
You need to call the new `await request.yar.commit(h)` before returning the redirect.

**Comment by suhanw** [comment]
Is this a new feature that's not available in yar@9.1.0?

**Comment by hueniverse** [comment]
Only in 10.1.0.
